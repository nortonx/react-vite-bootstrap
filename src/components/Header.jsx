import React from "react";
import { Link } from "react-router-dom";
const Header = () => {
  return(
    <div className="container header">
      <header className="d-flex flex-wrap justify-content-center py-3 mb-4 border-bottom">
        <a href="/" className="d-flex align-items-center mb-3 mb-md-0 me-md-auto text-dark text-decoration-none">Brand</a>
        <ul className="nav nav-pills">
          <li className="nav-item">
            <Link to="/" className="nav-link active">Home</Link>
          </li>
          <li className="nav-item">
            <Link to="/about" className="nav-link">About</Link>
          </li>
        </ul>
      </header>
    </div>
  )
}

export default Header;