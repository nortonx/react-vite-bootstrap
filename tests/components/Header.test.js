import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import Header from "../../src/components/Header"

describe("Header", () => {
  it("should render the component and match snapshot", async () => {
    render(<Header />, {wrapper: BrowserRouter})
    const element = await screen.findByTestId("header");
    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
  });
});