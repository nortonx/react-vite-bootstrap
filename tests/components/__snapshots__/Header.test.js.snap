// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component and match snapshot 1`] = `
<div
  class="container header"
  data-testid="header"
>
  <header
    class="d-flex flex-wrap justify-content-center py-3 mb-4 border-bottom"
  >
    <a
      class="d-flex align-items-center mb-3 mb-md-0 me-md-auto text-dark text-decoration-none"
      href="/"
    >
      Brand
    </a>
    <ul
      class="nav nav-pills"
    >
      <li
        class="nav-item"
      >
        <a
          activeclassname="active"
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          activeclassname="active"
          class="nav-link"
          href="/form"
        >
          Form
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          activeclassname="active"
          class="nav-link"
          href="/text-analyzer"
        >
          Text Analyzer
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          activeclassname="active"
          class="nav-link"
          href="/about"
        >
          About
        </a>
      </li>
    </ul>
  </header>
</div>
`;
